src/rnd.cpp

```diff
@@ -58,10 +58,10 @@
 //  Has a full period of 2^31 - 1.
 //  Returns integers in the range 1 to 2^31-1.
 
-#define RNG_M 2147483647L // m = 2^31 - 1
-#define RNG_A 16807L
-#define RNG_Q 127773L // m div a
-#define RNG_R 2836L   // m mod a
+constexpr long RNG_M = 0b10000000000000000000000000000000L - 1; // m = 2^31 - 1
+constexpr long RNG_A = 16807L;
+constexpr long RNG_Q = RNG_M / RNG_A; // m div a 127773L
+constexpr long RNG_R = RNG_M % RNG_A;   // m mod a 2836L
```

Comment:
Throughout the code I have been trying to use the standard types (`uint8_t`, 'int32_t`, etc). Is there a specific reason for using `long` here?